src/shared/condition.c

```diff
@@ -184,6 +185,8 @@ static int condition_test_credential(Condition *c, char **env) {
 typedef enum {
         /* Listed in order of checking. Note that some comparators are prefixes of others, hence the longest
          * should be listed first. */
+        ORDER_FNMATCH_EQUAL,
```

Comment:
maybe do it like this:

```c
_ORDER_FNMATCH_FIRST,
ORDER_FNMATCH_EQUAL = _ORDER_FNMATCH_FIRST,
ORDER_FNMATCH_UNEQUAL,
_ORDER_FNMATCH_LAST = ORDER_FNMATCH_UNEQUAL,
```

This then allows checking for the fnmatch operators generically like this:

```c
if (i >= _ORDER_FNMATCH_FIRST && i <= _ORDER_FNMATCH_LAST) …
```

Thus you don't have to hardcode which is the first or last, but have a generic way to reference it, and adding more is obvious.

Reply:
Very good idea! I've implemented this in the latest version.